x-pack/plugins/cases/common/api/cases/case.ts

```diff
@@ -38,6 +38,8 @@ const CaseBasicRt = rt.type({
   [caseTypeField]: CaseTypeRt,
   connector: CaseConnectorRt,
   settings: SettingsRt,
+  // TODO: should a user be able to update the class?
```

Comment:
For the moment we shouldn't do it. If in the feature we want to "transfer" a case to another solution we can enabled it. We can check this manually in the `update` and throw an error.

Reply:
Sounds good to me!